src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -569,6 +573,48 @@ protected void validateSequenceDictionaries() {
         }
     }
 
+    /**
+     * Register the reference file (and associated dictionary and index) to be downloaded to every node using Spark's
+     * copying mechanism ({@code SparkContext#addFile()}).
+     * @param ctx the Spark context
+     * @param referenceFile the reference file, can be a local file or a remote path
+     * @return the reference file name; the absolute path of the file can be found by a Spark task using {@code SparkFiles#get()}
+     */
+    protected String addReferenceFilesForSpark(JavaSparkContext ctx, String referenceFile) {
+        Path referencePath = IOUtils.getPath(referenceFile);
+        Path indexPath = ReferenceSequenceFileFactory.getFastaIndexFileName(referencePath);
+        Path dictPath = ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(referencePath);
+
+        ctx.addFile(referenceFile);
+        ctx.addFile(indexPath.toUri().toString());
+        ctx.addFile(dictPath.toUri().toString());
+
+        return referencePath.getFileName().toString();
+    }
+
+    /**
+     * Register the VCF file (and associated index) to be downloaded to every node using Spark's copying mechanism
+     * ({@code SparkContext#addFile()}).
+     * @param ctx the Spark context
+     * @param knownVariants the known variant (VCF) files, can be local files or remote paths
+     * @return the reference file name; the absolute path of the file can be found by a Spark task using {@code SparkFiles#get()}
+     */
+    protected List<String> addKnownSitesForSpark(JavaSparkContext ctx, List<String> knownVariants) {
```

Comment:
Looking at every usage of this method, it has to be called immediately before `JoinReadsWithVariants` methods are called. I would either make this a more generic like 'addVCFsToSpark` so it could be used for other purposes or I would explicitly make it part of `JoinReadsWithVariants` to avoid confusion about passing around the wrong List<String> objects when people need to distribute vcf files. Either way it looks to me that this method could be static anyway.

Reply:
Renamed and made static.